File: articles/knowledge-systems-and-scientific-collaboration/cpp/scientific_collaboration_metrics.cpp
#include <iostream>
#include <iomanip>

int main() {
    double objects = 14.0;
    double relationships = 16.0;

    std::cout << std::fixed << std::setprecision(3);
    std::cout << "Metadata coverage: " << 13.0 / objects << "\n";
    std::cout << "Provenance coverage: " << 13.0 / objects << "\n";
    std::cout << "Review context coverage: " << 11.0 / objects << "\n";
    std::cout << "Relationship traceability: " << 15.0 / relationships << "\n";
    std::cout << "Underspecified relationship risk: " << 1.0 / relationships << "\n";

    return 0;
}

File: articles/knowledge-systems-and-scientific-collaboration/fortran/scientific_collaboration_metrics.f90
program scientific_collaboration_metrics
  implicit none

  real :: objects
  real :: relationships

  objects = 14.0
  relationships = 16.0

  print *, "Scientific Collaboration Knowledge System Metrics"
  print *, "Metadata coverage: ", 13.0 / objects
  print *, "Provenance coverage: ", 13.0 / objects
  print *, "Review context coverage: ", 11.0 / objects
  print *, "Relationship traceability: ", 15.0 / relationships
  print *, "Underspecified relationship risk: ", 1.0 / relationships
end program scientific_collaboration_metrics
